api/src/main/java/org/openmrs/module/insuranceclaims/api/service/fhir/impl/FHIRClaimDiagnosisServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+package org.openmrs.module.insuranceclaims.api.service.fhir.impl;
+
+import org.hl7.fhir.dstu3.model.Claim;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.openmrs.Concept;
+import org.openmrs.module.fhir.api.util.BaseOpenMRSDataUtil;
+import org.openmrs.module.fhir.api.util.FHIRUtils;
+import org.openmrs.module.insuranceclaims.api.dao.InsuranceClaimDiagnosisDao;
+import org.openmrs.module.insuranceclaims.api.model.InsuranceClaim;
+import org.openmrs.module.insuranceclaims.api.model.InsuranceClaimDiagnosis;
+import org.openmrs.module.insuranceclaims.api.service.fhir.FHIRClaimDiagnosisService;
+
+import javax.transaction.Transactional;
+import java.util.LinkedList;
+import java.util.List;
+
+public class FHIRClaimDiagnosisServiceImpl implements FHIRClaimDiagnosisService {
+
+    private InsuranceClaimDiagnosisDao diagnosisDao;
+
+    public FHIRClaimDiagnosisServiceImpl() {
+    }
+
+    public FHIRClaimDiagnosisServiceImpl(InsuranceClaimDiagnosisDao diagnosisDao) {
```

Comment:
I see that in xml you autowired in using property - setter and constructor are redundant in this case